amy/extrequests/views.py

```diff
@@ -726,53 +726,30 @@ def _match_training_request_to_person(
         )
         return False
 
-    # Create new style consents according to selection from the training request
-    # form.
-    try:
-        option_type = (
-            TermOptionChoices.AGREE
-            if training_request.data_privacy_agreement
-            else TermOptionChoices.DECLINE
-        )
-
-        reconsent_for_term_option_type(
-            term_key=TermEnum.PRIVACY_POLICY,
-            term_option_type=option_type,
-            person=training_request.person,
-        )
-
-    except (Term.DoesNotExist, TermOption.DoesNotExist):
-        logger.warning(
-            f"Either Term {TermEnum.PRIVACY_POLICY} or its term option was not found, "
-            "can't proceed."
-        )
-        messages.error(
-            request,
-            f"Error when setting person's consents. Term {TermEnum.PRIVACY_POLICY} or "
-            "related term option may not exist.",
-        )
-        return False
-
-    try:
-        # If they created a training request, we assume they agreed to "may contact"
-        # consent.
-        reconsent_for_term_option_type(
-            term_key=TermEnum.MAY_CONTACT,
-            term_option_type=TermOptionChoices.AGREE,
-            person=training_request.person,
-        )
-
-    except (Term.DoesNotExist, TermOption.DoesNotExist):
-        logger.warning(
-            f"Either Term {TermEnum.MAY_CONTACT} or its term option was not found, "
-            "can't proceed."
-        )
-        messages.error(
-            request,
-            f"Error when setting person's consents. Term {TermEnum.MAY_CONTACT} or "
-            "related term option may not exist.",
-        )
-        return False
+    # Create new style consents based on the training request consents used in the
+    # training request form.
+    training_request_consents = TrainingRequestConsent.objects.filter(
```

Comment:
> After this change/migration, we will have some training requests in the database which don't have any TrainingRequestConsents set. Am I right in understanding that if a Person is created from such a request, they will instead provide these consents when they first log in to AMY?

I have just tested this and you are correct.

> Locally I get an error:

Yes, this is because it's trying to confirm GH handle with GH API. You can erase the handle from request by editing it and you should be able to continue.

You can also create a token for your own use here: https://github.com/settings/tokens
Use `AMY_GITHUB_API_TOKEN` envvar to set the token.